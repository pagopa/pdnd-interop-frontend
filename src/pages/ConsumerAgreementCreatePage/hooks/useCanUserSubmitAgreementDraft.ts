import { AgreementQueries } from '@/api/agreement'
import { AttributeQueries } from '@/api/attribute'
import { EServiceQueries } from '@/api/eservice'
import { useJwt } from '@/hooks/useJwt'
import { hasAllEServiceAttributes, remapDescriptorAttributes } from '@/utils/attribute.utils'
import React from 'react'

export default function useCanUserSubmitAgreementDraft(agreementId: string) {
  const { jwt } = useJwt()
  const { data: agreement } = AgreementQueries.useGetSingle(agreementId, { suspense: false })
  // This should not stay here, waiting to get the attributes from the agreement itself
  const { data: descriptor } = EServiceQueries.useGetDescriptorCatalog(
    agreement?.eservice.id as string,
    agreement?.descriptorId as string,
    { enabled: !!(agreement?.eservice.id && agreement?.descriptorId), suspense: false }
  )

  const [{ data: ownedCertified }, , { data: ownedDeclared }] = AttributeQueries.useGetListParty(
    jwt?.organizationId,
    agreement?.producer.id,
    {
      suspense: false,
    }
  )

  return React.useMemo(() => {
    if (!agreement || !descriptor || !ownedCertified || !ownedDeclared) return false

    const isProviderSameAsSubscriber = agreement.consumer.id === agreement.producer.id
    const remapedEServiceAttributes = remapDescriptorAttributes(descriptor.attributes)

    const hasAllCertifiedAttributes =
      agreement?.state !== 'MISSING_CERTIFIED_ATTRIBUTES' &&
      hasAllEServiceAttributes(
        'certified',
        ownedCertified.attributes,
        remapedEServiceAttributes.certified
      )

    const hasAllDeclaredAttributes = hasAllEServiceAttributes(
      'declared',
      ownedDeclared.attributes,
      remapedEServiceAttributes.declared
    )

    /**
     * If the provider is the same as the subscriber, we don't need to check for attributes
     */
    return isProviderSameAsSubscriber || (hasAllCertifiedAttributes && hasAllDeclaredAttributes)
  }, [agreement, descriptor, ownedCertified, ownedDeclared])
}
