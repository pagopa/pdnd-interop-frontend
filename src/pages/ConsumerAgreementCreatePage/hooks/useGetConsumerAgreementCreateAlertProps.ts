import type { Agreement } from '@/api/api.generatedTypes'
import { useDialog } from '@/stores'
import { isNewEServiceVersionAvailable } from '@/utils/agreement.utils'
import type { AlertProps } from '@mui/material'
import { useTranslation } from 'react-i18next'

export function useGetConsumerAgreementCreateAlertProps(agreement: Agreement | undefined):
  | {
      severity: AlertProps['severity']
      content: string
      action?: VoidFunction
    }
  | undefined {
  const { t } = useTranslation('agreement')

  const { openDialog } = useDialog()

  if (!agreement || agreement.state !== 'DRAFT') return undefined

  if (isNewEServiceVersionAvailable(agreement)) {
    return {
      severity: 'warning',
      content: t('edit.newVersionAlert'),
    }
  }

  const hasSetContactEmail = agreement && !!agreement?.consumer.contactMail?.address
  if (!hasSetContactEmail) {
    return {
      severity: 'warning',
      content: t('edit.noContactEmailAlert'),
      action: () => openDialog({ type: 'setTenantMail' }),
    }
  }

  const isDelegated = agreement && agreement.delegation
  if (isDelegated) {
    return {
      severity: 'info',
      content: t('edit.delegatedAttributesAlert', { delegatorName: agreement.consumer.name }),
    }
  }
}
