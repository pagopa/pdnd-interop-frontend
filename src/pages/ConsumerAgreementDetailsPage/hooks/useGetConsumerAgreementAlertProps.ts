import type { Agreement } from '@/api/api.generatedTypes'
import { PurposeQueries } from '@/api/purpose'
import type { Link, RouteKey } from '@/router'
import type { AlertProps } from '@mui/material'
import { useQuery } from '@tanstack/react-query'
import { useTranslation } from 'react-i18next'

export function useGetConsumerAgreementAlertProps(agreement: Agreement):
  | {
      severity: AlertProps['severity']
      content: string
      link?: {
        to: RouteKey
        params?: React.ComponentProps<typeof Link>['params']
        options?: React.ComponentProps<typeof Link>['options']
      }
    }
  | undefined {
  const { t } = useTranslation('agreement')

  const { data: purposes } = useQuery({
    ...PurposeQueries.getConsumersList({
      limit: 50,
      offset: 0,
      producersIds: [agreement.producer.id],
      eservicesIds: [agreement.eservice.id],
    }),
    enabled: agreement.state === 'ACTIVE',
  })

  const suspendedBy = (() => {
    if (agreement.state !== 'SUSPENDED') return undefined
    if (agreement.suspendedByProducer) return 'byProducer'
    if (agreement.suspendedByConsumer) return 'byConsumer'
    if (agreement.suspendedByPlatform) return 'byPlatform'
  })()

  if (suspendedBy) {
    return {
      severity: 'error',
      content: t(`consumerRead.suspendedAlert.${suspendedBy}`),
    }
  }

  if (agreement.state === 'MISSING_CERTIFIED_ATTRIBUTES') {
    return {
      severity: 'warning',
      content: t('consumerRead.missingCertifiedAttributesAlert'),
    }
  }

  const isWithoutPurposes =
    purposes?.results.length === 0 ||
    purposes?.results.every((purpose) => purpose.currentVersion?.state === 'ARCHIVED')

  if (isWithoutPurposes) {
    return {
      severity: 'info',
      content: t('consumerRead.noPurposeAlert'),
      link: {
        to: 'SUBSCRIBE_PURPOSE_CREATE',
      },
    }
  }
}
