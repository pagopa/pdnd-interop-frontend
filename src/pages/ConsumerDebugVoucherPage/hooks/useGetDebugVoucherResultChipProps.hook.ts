import { useTranslation } from 'react-i18next'
import type { TokenGenerationValidationEntry } from '../types/debug-voucher.types'
import type { ChipProps } from '@mui/material'

export function useGetDebugVoucherResultChipProps(
  step?: TokenGenerationValidationEntry
): ChipProps | undefined {
  const { t } = useTranslation('voucher', { keyPrefix: 'consumerDebugVoucher.result' })

  if (!step) {
    return
  }

  if (step.result === 'PASSED') {
    return {
      label: t('chipLabel.passed'),
      color: 'success',
    }
  }

  if (step.result === 'SKIPPED') {
    return {
      label: t('chipLabel.skipped'),
      color: 'warning',
    }
  }

  if (step.result === 'FAILED') {
    return {
      label: t('chipLabel.failed', {
        count: step.failures.length,
      }),
      color: 'error',
    }
  }

  throw new Error('Unknown step result:', step.result)
}
