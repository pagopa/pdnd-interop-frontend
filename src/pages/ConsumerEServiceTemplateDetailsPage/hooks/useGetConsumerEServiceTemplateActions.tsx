import type { EServiceTemplateVersionState } from '@/api/api.generatedTypes'
import { useNavigate } from '@/router'
import { useTranslation } from 'react-i18next'
import type { ActionItemButton } from '@/types/common.types'
import { AuthHooks } from '@/api/auth'
import FiberNewIcon from '@mui/icons-material/FiberNew'
import { EServiceQueries } from '@/api/eservice'
import { useQuery } from '@tanstack/react-query'

export function useGetConsumerEServiceTemplateActions(
  eServiceTemplateId: string,
  canBeInstantiated: boolean,
  activeVersionState?: EServiceTemplateVersionState | undefined
): { actions: Array<ActionItemButton> } {
  const { t } = useTranslation('template', { keyPrefix: 'actions' })

  const { isAdmin, isOperatorAPI } = AuthHooks.useJwt()
  const navigate = useNavigate()

  const state = activeVersionState ?? 'DRAFT'

  // Only admin and operatorAPI can see actions
  if (!isAdmin && !isOperatorAPI) return { actions: [] }

  const handleCreateEServiceFromTemplate = () => {
    navigate('PROVIDE_ESERVICE_FROM_TEMPLATE_CREATE', {
      params: { eServiceTemplateId: eServiceTemplateId },
    })
  }

  const tooltipLabel = t('createNewEServiceInstanceDisabled')
    .split('\n')
    .map((line, idx) => (
      <span key={idx}>
        {line}
        <br />
      </span>
    ))

  const newEServiceFromTemplateAction: ActionItemButton = {
    action: handleCreateEServiceFromTemplate,
    label: t('createNewEServiceInstance'),
    icon: FiberNewIcon,
    disabled: !canBeInstantiated,
  }

  const publishedConsumerActions = [newEServiceFromTemplateAction]

  const adminActions: Record<EServiceTemplateVersionState, Array<ActionItemButton>> = {
    PUBLISHED: isAdmin ? publishedConsumerActions : [],
    DRAFT: [],
    SUSPENDED: [],
    DEPRECATED: [],
  }

  const operatorAPIActions: Record<EServiceTemplateVersionState, Array<ActionItemButton>> = {
    PUBLISHED: [],
    DRAFT: [],
    SUSPENDED: [],
    DEPRECATED: [],
  }

  const availableAction = isAdmin ? adminActions[state] : operatorAPIActions[state]

  return { actions: availableAction }
}
