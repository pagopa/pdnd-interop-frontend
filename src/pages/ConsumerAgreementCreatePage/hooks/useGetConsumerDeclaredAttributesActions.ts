import { isAttributeOwned } from '@/utils/attribute.utils'
import { useTranslation } from 'react-i18next'
import { useJwt } from '@/hooks/useJwt'
import { AttributeMutations } from '@/api/attribute'
import { useConsumerAgreementCreateContentContext } from '../ConsumerAgreementCreateContentContext'

export const useGetConsumerDeclaredAttributesActions = () => {
  const { t } = useTranslation('agreement', { keyPrefix: 'read.attributes' })
  const { isAdmin } = useJwt()
  const { partyAttributes, agreement } = useConsumerAgreementCreateContentContext()

  const { mutate: declareAttribute } = AttributeMutations.useDeclarePartyAttribute()

  return (attributeId: string) => {
    // The user can declare his own attributes only in the agreement create/edit view...
    if (!agreement || !isAdmin) return []
    const isDeclared = isAttributeOwned('declared', attributeId, partyAttributes?.declared ?? [])
    // ... only if it is not alread declared
    if (isDeclared) return []
    // ... and only if the agreement is active, draft or suspended
    if (!['DRAFT'].includes(agreement.state)) return []

    const handleDeclareAttribute = (attributeId: string) => {
      declareAttribute({
        id: attributeId,
      })
    }

    return [
      {
        label: t('declared.actions.declare'),
        action: handleDeclareAttribute,
      },
    ]
  }
}
