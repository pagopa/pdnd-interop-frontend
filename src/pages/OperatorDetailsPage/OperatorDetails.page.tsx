import React from 'react'
import { ClientMutations, ClientQueries } from '@/api/client'
import { PageBottomActionsContainer, PageContainer } from '@/components/layout/containers'
import { RouterLink, useCurrentRoute, useRouteParams } from '@/router'
import {
  OperatorGeneralInfoSection,
  OperatorGeneralInfoSectionSkeleton,
} from './components/OperatorGeneralInfoSection'
import { OperatorKeysSection, OperatorKeysSectionSkeleton } from './components/OperatorKeysSection'
import type { ActionItem } from '@/types/common.types'
import { useJwt } from '@/hooks/useJwt'
import { useTranslation } from 'react-i18next'
import { formatTopSideActions } from '@/utils/common.utils'
import { Grid } from '@mui/material'
import { useClientKind } from '@/hooks/useClientKind'

const OperatorDetailsPage: React.FC = () => {
  const { isAdmin } = useJwt()
  const { mode } = useCurrentRoute()
  const clientKind = useClientKind()
  const { t } = useTranslation('user')
  const { mutate: removeOperatorFromClient } = ClientMutations.useRemoveOperator()

  const { clientId: clientId, operatorId } = useRouteParams<
    'SUBSCRIBE_INTEROP_M2M_CLIENT_OPERATOR_EDIT' | 'SUBSCRIBE_CLIENT_OPERATOR_EDIT'
  >()
  const { data: operator, isLoading } = ClientQueries.useGetSingleOperator(operatorId, {
    suspense: false,
  })

  const operatorFullname = `${operator?.name} ${operator?.familyName}`

  const actions: Array<ActionItem> = []

  if (mode === 'consumer' && isAdmin) {
    actions.push({
      action: removeOperatorFromClient.bind(null, { clientId, relationshipId: operatorId }),
      label: t('actions.removeFromClient'),
    })
  }

  const topSideActions = formatTopSideActions(actions, { variant: 'contained' })
  const backToOperatorsListRouteKey =
    clientKind === 'API' ? 'SUBSCRIBE_INTEROP_M2M_CLIENT_EDIT' : 'SUBSCRIBE_CLIENT_EDIT'

  return (
    <PageContainer isLoading={isLoading} title={operatorFullname} topSideActions={topSideActions}>
      <Grid spacing={2} container>
        <Grid item xs={7}>
          <React.Suspense fallback={<OperatorGeneralInfoSectionSkeleton />}>
            <OperatorGeneralInfoSection operatorId={operatorId} />
          </React.Suspense>
        </Grid>
        <Grid item xs={5}>
          {operator?.product.role === 'security' && (
            <React.Suspense fallback={<OperatorKeysSectionSkeleton />}>
              <OperatorKeysSection clientId={clientId} operatorId={operatorId} />
            </React.Suspense>
          )}
        </Grid>
      </Grid>
      <PageBottomActionsContainer>
        <RouterLink
          as="button"
          variant="outlined"
          to={backToOperatorsListRouteKey}
          params={{ clientId }}
          options={{ urlParams: { tab: 'clientOperators' } }}
        >
          {t('backToMemberListBtn')}
        </RouterLink>
      </PageBottomActionsContainer>
    </PageContainer>
  )
}

export default OperatorDetailsPage
