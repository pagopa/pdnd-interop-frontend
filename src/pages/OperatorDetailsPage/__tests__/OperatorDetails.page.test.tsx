import React from 'react'
import { BACKEND_FOR_FRONTEND_URL } from '@/config/env'
import {
  mockUseCurrentRoute,
  mockUseJwt,
  renderWithApplicationContext,
  setupQueryServer,
} from '@/utils/testing.utils'
import { createMockSelfCareUser } from '__mocks__/data/user.mocks'
import * as useClientKindHook from '@/hooks/useClientKind'
import * as router from '@/router'
import { vi } from 'vitest'
import OperatorDetailsPage from '../OperatorDetails.page'
import { waitFor } from '@testing-library/react'

const queryServer = setupQueryServer([
  {
    url: `${BACKEND_FOR_FRONTEND_URL}/relationships/:relationshipId`,
    result: createMockSelfCareUser({ product: { role: 'security' } }),
  },
])

beforeAll(() => queryServer.listen())
afterEach(() => queryServer.resetHandlers())
afterAll(() => queryServer.close())

const useClientKindSpy = vi.spyOn(useClientKindHook, 'useClientKind')
vi.spyOn(router, 'useRouteParams').mockReturnValue({
  clientId: 'clientId',
  operatorId: 'operatorId',
})

describe('OperatorGeneralInfoSection', () => {
  it('should match the snapshot (API)', async () => {
    useClientKindSpy.mockReturnValue('API')
    mockUseCurrentRoute({ mode: 'provider' })
    const screen = renderWithApplicationContext(<OperatorDetailsPage />, {
      withReactQueryContext: true,
      withRouterContext: true,
    })
    expect(screen.baseElement).toMatchSnapshot('loading state')
    await waitFor(() => screen.getByRole('heading', { name: 'Mario Rossi' }))
    expect(screen.baseElement).toMatchSnapshot('full state')
  })

  it('should match the snapshot (CONSUMER)', async () => {
    useClientKindSpy.mockReturnValue('CONSUMER')
    mockUseCurrentRoute({ mode: 'provider' })
    const screen = renderWithApplicationContext(<OperatorDetailsPage />, {
      withReactQueryContext: true,
      withRouterContext: true,
    })
    expect(screen.baseElement).toMatchSnapshot('loading state')
    await waitFor(() => screen.getByRole('heading', { name: 'Mario Rossi' }))
    expect(screen.baseElement).toMatchSnapshot('full state')
  })

  it("should render the 'remove from client' action if user is admin and the context is 'consumer'", async () => {
    useClientKindSpy.mockReturnValue('CONSUMER')
    mockUseCurrentRoute({ mode: 'consumer' })
    mockUseJwt({ isAdmin: true })
    const screen = renderWithApplicationContext(<OperatorDetailsPage />, {
      withReactQueryContext: true,
      withRouterContext: true,
    })
    await waitFor(() => screen.getByRole('heading', { name: 'Mario Rossi' }))
    expect(screen.getByRole('button', { name: 'actions.removeFromClient' })).toBeInTheDocument()
  })

  it("should not render the 'remove from client' action if user is not admin", async () => {
    useClientKindSpy.mockReturnValue('CONSUMER')
    mockUseCurrentRoute({ mode: 'consumer' })
    mockUseJwt({ isAdmin: false })
    const screen = renderWithApplicationContext(<OperatorDetailsPage />, {
      withReactQueryContext: true,
      withRouterContext: true,
    })
    await waitFor(() => screen.getByRole('heading', { name: 'Mario Rossi' }))
    expect(
      screen.queryByRole('button', { name: 'actions.removeFromClient' })
    ).not.toBeInTheDocument()
  })

  it("should not render the 'remove from client' action if context is provider", async () => {
    useClientKindSpy.mockReturnValue('CONSUMER')
    mockUseCurrentRoute({ mode: 'provider' })
    mockUseJwt({ isAdmin: false })
    const screen = renderWithApplicationContext(<OperatorDetailsPage />, {
      withReactQueryContext: true,
      withRouterContext: true,
    })
    await waitFor(() => screen.getByRole('heading', { name: 'Mario Rossi' }))
    expect(
      screen.queryByRole('button', { name: 'actions.removeFromClient' })
    ).not.toBeInTheDocument()
  })
})
