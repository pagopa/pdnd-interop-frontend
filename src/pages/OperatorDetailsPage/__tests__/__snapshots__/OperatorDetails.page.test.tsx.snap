// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OperatorGeneralInfoSection > should match the snapshot (API) > full state 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="css-1mure3k-MuiStack-root"
      >
        <div
          class="MuiBox-root css-1rr4qq7"
        >
          <h1
            class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
          >
            Mario Rossi
          </h1>
        </div>
        <div
          class="css-1yo1jo6-MuiStack-root"
        />
      </div>
      <div
        class="MuiBox-root css-h5fkc8"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-uqwprf-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w5x330-MuiPaper-root"
            >
              <div
                class="css-nen11g-MuiStack-root"
              >
                <h2
                  class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
                >
                  generalInformations
                </h2>
              </div>
              <div
                class="MuiBox-root css-1yuhvjn"
              >
                <div
                  class="css-1p5q5e5-MuiStack-root"
                >
                  <div
                    class="css-88r4rx-MuiStack-root"
                  >
                    <div
                      class="MuiBox-root css-iuqs4u"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                      >
                        taxCodeField.label
                      </p>
                    </div>
                    <div
                      class="MuiBox-root css-1rr4qq7"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
                      >
                        MRORSSR78M21B354T
                      </p>
                    </div>
                  </div>
                  <div
                    class="css-19qjufr-MuiStack-root"
                  >
                    <div
                      class="MuiBox-root css-iuqs4u"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                      >
                        productRoleField.label
                      </p>
                    </div>
                    <div
                      class="MuiBox-root css-1rr4qq7"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
                      >
                        userProductRole.security
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-4i8vzz-MuiSkeleton-root"
              style="height: 107px;"
            />
          </div>
        </div>
        <div
          class="css-fhd9dx-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            options="[object Object]"
            tabindex="0"
            type="button"
          >
            backToMemberListBtn
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`OperatorGeneralInfoSection > should match the snapshot (API) > loading state 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="css-1mure3k-MuiStack-root"
      >
        <div
          class="MuiBox-root css-1rr4qq7"
        >
          <h1
            class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
            />
          </h1>
          <p
            class="MuiTypography-root MuiTypography-body1 css-u9n5y4-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
            />
          </p>
        </div>
        <div
          class="css-1yo1jo6-MuiStack-root"
        />
      </div>
      <div
        class="MuiBox-root css-h5fkc8"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-uqwprf-MuiGrid-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-4i8vzz-MuiSkeleton-root"
              style="height: 235px;"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
          />
        </div>
        <div
          class="css-fhd9dx-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            options="[object Object]"
            tabindex="0"
            type="button"
          >
            backToMemberListBtn
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`OperatorGeneralInfoSection > should match the snapshot (CONSUMER) > full state 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="css-1mure3k-MuiStack-root"
      >
        <div
          class="MuiBox-root css-1rr4qq7"
        >
          <h1
            class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
          >
            Mario Rossi
          </h1>
        </div>
        <div
          class="css-1yo1jo6-MuiStack-root"
        />
      </div>
      <div
        class="MuiBox-root css-h5fkc8"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-uqwprf-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w5x330-MuiPaper-root"
            >
              <div
                class="css-nen11g-MuiStack-root"
              >
                <h2
                  class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
                >
                  generalInformations
                </h2>
              </div>
              <div
                class="MuiBox-root css-1yuhvjn"
              >
                <div
                  class="css-1p5q5e5-MuiStack-root"
                >
                  <div
                    class="css-88r4rx-MuiStack-root"
                  >
                    <div
                      class="MuiBox-root css-iuqs4u"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                      >
                        taxCodeField.label
                      </p>
                    </div>
                    <div
                      class="MuiBox-root css-1rr4qq7"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
                      >
                        MRORSSR78M21B354T
                      </p>
                    </div>
                  </div>
                  <div
                    class="css-19qjufr-MuiStack-root"
                  >
                    <div
                      class="MuiBox-root css-iuqs4u"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                      >
                        productRoleField.label
                      </p>
                    </div>
                    <div
                      class="MuiBox-root css-1rr4qq7"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
                      >
                        userProductRole.security
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-4i8vzz-MuiSkeleton-root"
              style="height: 107px;"
            />
          </div>
        </div>
        <div
          class="css-fhd9dx-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            options="[object Object]"
            tabindex="0"
            type="button"
          >
            backToMemberListBtn
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`OperatorGeneralInfoSection > should match the snapshot (CONSUMER) > loading state 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="css-1mure3k-MuiStack-root"
      >
        <div
          class="MuiBox-root css-1rr4qq7"
        >
          <h1
            class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
          >
            Mario Rossi
          </h1>
        </div>
        <div
          class="css-1yo1jo6-MuiStack-root"
        />
      </div>
      <div
        class="MuiBox-root css-h5fkc8"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-uqwprf-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w5x330-MuiPaper-root"
            >
              <div
                class="css-nen11g-MuiStack-root"
              >
                <h2
                  class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
                >
                  generalInformations
                </h2>
              </div>
              <div
                class="MuiBox-root css-1yuhvjn"
              >
                <div
                  class="css-1p5q5e5-MuiStack-root"
                >
                  <div
                    class="css-88r4rx-MuiStack-root"
                  >
                    <div
                      class="MuiBox-root css-iuqs4u"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                      >
                        taxCodeField.label
                      </p>
                    </div>
                    <div
                      class="MuiBox-root css-1rr4qq7"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
                      >
                        MRORSSR78M21B354T
                      </p>
                    </div>
                  </div>
                  <div
                    class="css-19qjufr-MuiStack-root"
                  >
                    <div
                      class="MuiBox-root css-iuqs4u"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                      >
                        productRoleField.label
                      </p>
                    </div>
                    <div
                      class="MuiBox-root css-1rr4qq7"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
                      >
                        userProductRole.security
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-4i8vzz-MuiSkeleton-root"
              style="height: 107px;"
            />
          </div>
        </div>
        <div
          class="css-fhd9dx-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            options="[object Object]"
            tabindex="0"
            type="button"
          >
            backToMemberListBtn
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
